File: csharp/phone-number/PhoneNumber.cs
using System;
using System.Text.RegularExpressions;

public class PhoneNumber
{
    public static string Clean(string number)
    {
        number = Regex.Replace(number, @"[^\d]", "");

        var match = Regex.Match(number, @"^1?(?<ac>[2-9]\d{2})(?<ec>[2-9]\d{2})(?<sn>\d{4})$");
        if (!match.Success)
        {
            throw new ArgumentException();
        }

        return $"{match.Groups["ac"].Value}{match.Groups["ec"].Value}{match.Groups["sn"].Value}";
    }
}

File: javascript/phone-number/phone-number.js
export default class PhoneNumber {
    constructor(phoneNumber) {
        this._nanpRegexp = /^1?([2-9]\d{2})([2-9]\d{2})(\d{4})$/;

        this._phoneNumber = this._getNumber(phoneNumber);
    }

    number() {
        return this._phoneNumber;
    }

    _getNumber(phoneNumber) {
        const match = phoneNumber.replace(/[\(\)\-\s\.]/g, '').match(this._nanpRegexp);
        
        return match !== null
            ? `${match[1]}${match[2]}${match[3]}`
            : null;
    }
}
